File: apps/ios/Barns/BarnsApp/Domain/UseCases/Care/GetCareLogsUseCase.swift
/// Returns local care logs, optionally filtered by product item.
struct GetCareLogsUseCase {
    private let repository: CareRepository

    init(repository: CareRepository) {
        self.repository = repository
    }

    func execute(productItemId: ProductItem.ID? = nil) async throws -> [CareLog] {
        try await repository.careLogs(for: productItemId)
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/care/GetCareLogsUseCase.kt
package com.barns.app.domain.usecase.care

import com.barns.app.domain.model.CareLog
import com.barns.app.domain.repository.CareRepository

/**
 * Returns local care logs, optionally filtered by product item.
 */
class GetCareLogsUseCase(
    private val repository: CareRepository,
) {
    suspend fun execute(productItemId: String? = null): List<CareLog> =
        repository.careLogs(productItemId)
}
